### PR TITLE
Set new XFEM node processor ids individually

### LINKED ISSUE #11500: Fix assignment of processor_id in newly created XFEM nodes
## Rationale

Recent changes to libMesh broke the XFEM `init_solution_propagation` test. 
Libmesh Issue https://github.com/libMesh/libmesh/issues/1688
It appears that this issue is because XFEM is creating new nodes that are not attached to any on-processor elements in some cases.

## Description

This is what we do in XFEM: if an element is cut, we delete it, and replace it with two new overlapping elements. Those elements are connected to both the original nodes and new nodes that are created at the same locations as the original nodes. The new elements are set to have the same processor id as the original one that we deleted, and the new nodes are set to have the same processor id as the ones that they are duplicated from. There are scenarios where the newly-created nodes may not be connected to any elements on-processor. 

## Impact

Minor changes to XFEM.C

### PR DESCRIPTION
**Pull request by roystgnr**
It looks like we can't safely set_node_processor_ids() en masse like
we tried in #11502, because that can change processor ids on existing
nodes in ways which confuse the libMesh DofMap.  Setting only the ids
on new nodes gets us passing module CI tests for me.

Fixes #11500 in my tests.  We shouldn't merge this until @jwpeterson or someone confirms that, though, because I still haven't figured out how I got false negatives when testing #11502.

I'm not confident this is the best possible fix, but this regression is blocking a lot of other work and at this point *any* fix is better than nothing.

### CODE REVIEW

**Comment by jwpeterson** [comment]
No offense but I'm going to try this branch locally with libmesh master before merging 😉

**Review by bwspenc — APPROVED** [review]
This looks good to me (assuming it passes with the latest libmesh).  Nice and simple.  Thanks for the help with this, @roystgnr!

**Comment by bwspenc** [comment]
@jwpeterson I'm going to defer to you to merge this after you try it out.

**Review by jwpeterson — APPROVED** [review]
This also gets everything passing for me 👍

**Pull request merged**
